android/titanium/src/java/org/appcelerator/titanium/TiFileProxy.java

```diff
@@ -153,6 +157,15 @@ public boolean createDirectory(@Kroll.argument(optional=true) Object arg)
 	}
 
 	@Kroll.method
+	public boolean createFile()
+	{
+		Context context = TiApplication.getInstance().getApplicationContext();
```

Comment:
`return tbf.createFile()` would mean `TiBaseFile`'s `createFile` method which would simply return false everytime (see [tbf.createFile).](https://github.com/appcelerator/titanium_mobile/pull/8514/files#diff-0c93a7953c3607b38b61844d828ba1a4R193) To start from the beginning, when the url is passed to the `TiFileProxy`, `tbf` is instantiated to a child class `TiResourceFile` which is a sibling to `TiFile`. In order to use `TiFile.createFile` I needed to convert `tbf` to `TiFile` and to do this I needed a path. `tbf.getNativePath` would throw errors stating that directory and file did not exist. Even using `mkdirs` or `mkdir` didn't fix this so I used `ContextWrapper` instead. It doesn't affect other url's and also `applicationDataDirectory`, `resourcesDirectory`, etc would return the same url in end when it is parsed in `TiFileFactory`. (see [storage locations](http://docs.appcelerator.com/platform/latest/#!/guide/Filesystem_Access_and_Storage)